### PR TITLE
*: Fix failed cases in sqllogictest-expr

### PR DESCRIPTION
**Pull request by shenli**
@siddontang @qiuyesuifeng 
pass sqllogictest-expr

### CODE REVIEW

File: util/types/field_type.go [hunk 1]
```diff
@@ -121,3 +121,15 @@ type DataItem struct {
 func (di *DataItem) String() string {
 	return fmt.Sprintf("%s", di.Data)
 }
+
+func GetRealData(d interface{}) interface{} {
```
Comment (ngaut) [inline util/types/field_type.go]: s/GetRealData/RawData/g

**Comment by siddontang** [comment]
How about group_concat, concat, concat_ws in builtin.

**Comment by shenli** [comment]
PTAL @siddontang @ngaut @qiuyesuifeng

File: plan/plans/final.go [hunk 2]
```diff
@@ -55,6 +56,14 @@ func (r *SelectFinalPlan) Filter(ctx context.Context, expr expression.Expression
 	return r, false, nil
 }
 
+func getInternalData(data interface{}) interface{} {
```
Comment (ngaut) [inline plan/plans/final.go]: Remove getInternalData function

**Comment by shenli** [comment]
PTAL

**Comment by ngaut** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by siddontang** [comment]
I think you can get raw data in Call Eval totally, so that every builtin function doesn't need to handle DataItem again.

**Comment by qiuyesuifeng** [comment]
@siddontang  PTAL

**Comment by shenli** [comment]
@siddontang No, in the next step, I will use the type information in each builtin.

**Comment by siddontang** [comment]
miss handing DataItem in string control builtin functions.

**Comment by shenli** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Pull request merged**